StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/Helpers/FixAllContextHelper.cs

```diff
@@ -0,0 +1,267 @@
+                        {
+                            types = ImmutableArray.Create(analyzerType);
+                        }
+
+                        analyzers[descriptor.Id] = types;
+                    }
+                }
+            }
+
+            return analyzers.ToImmutable();
+        }
+
+        private static ImmutableArray<DiagnosticAnalyzer> GetDiagnosticAnalyzersForContext(FixAllContext fixAllContext)
+        {
+            return DiagnosticAnalyzers
+                .Where(x => fixAllContext.DiagnosticIds.Contains(x.Key))
+                .SelectMany(x => x.Value)
+                .Distinct()
+                .Select(type => (DiagnosticAnalyzer)Activator.CreateInstance(type))
+                .ToImmutableArray();
+        }
+
+        private static async Task<ImmutableArray<Diagnostic>> GetAllDiagnosticsAsync(FixAllContext fixAllContext, Project project, ImmutableArray<DiagnosticAnalyzer> analyzers)
+        {
+            bool includeCompilerDiagnostics = fixAllContext.DiagnosticIds.Any(x => x.StartsWith("CS", StringComparison.Ordinal));
+            var compilation = await project.GetCompilationAsync(fixAllContext.CancellationToken).ConfigureAwait(false);
+
+            var diagnostics = ImmutableArray<Diagnostic>.Empty;
+            if (analyzers.Any())
+            {
+                var compilationWithAnalyzers = compilation.WithAnalyzers(analyzers, project.AnalyzerOptions, fixAllContext.CancellationToken);
+                if (GetAnalyzerSyntaxDiagnosticsAsync != null && GetAnalyzerSemanticDiagnosticsAsync != null)
+                {
+                    // This whole block is workaround code for issues with Roslyn 1.1...
+                    compilationWithAnalyzers.Compilation.GetDeclarationDiagnostics(fixAllContext.CancellationToken);
+
+                    var tasks = new List<Task<ImmutableArray<Diagnostic>>>();
+                    foreach (var document in project.Documents)
+                    {
+                        var syntaxTree = await document.GetSyntaxTreeAsync(fixAllContext.CancellationToken).ConfigureAwait(false);
+                        var syntaxDiagnostics = await GetAnalyzerSyntaxDiagnosticsAsync(compilationWithAnalyzers, syntaxTree, fixAllContext.CancellationToken).ConfigureAwait(false);
+                        diagnostics = diagnostics.AddRange(syntaxDiagnostics);
+
+                        var semanticModel = await document.GetSemanticModelAsync(fixAllContext.CancellationToken).ConfigureAwait(false);
+                        var semanticDiagnostics = await GetAnalyzerSemanticDiagnosticsAsync(compilationWithAnalyzers, semanticModel, default(TextSpan?), fixAllContext.CancellationToken).ConfigureAwait(false);
+                        diagnostics = diagnostics.AddRange(semanticDiagnostics);
+                    }
+                }
+                else
+                {
+                    diagnostics = await compilationWithAnalyzers.GetAnalyzerDiagnosticsAsync().ConfigureAwait(false);
```

Comment:
I would go back to `FixAllContext.GetAllDiagnosticsAsync` - it will give better performance compared to `CompilationWithAnalyzers` when background analysis has already computed and cached the diagnostics.
Can you also let us know how to run your fixall perf tests? We would surely want to validate that the issues in `FixAllContext.GetAllDiagnosticsAsync` are fully addressed in 1.2